pkg/backup/backuputils/utils.go

```diff
@@ -82,3 +86,39 @@ func AppendPaths(uris []string, tailDir ...string) ([]string, error) {
 	}
 	return retval, nil
 }
+
+// EncodeDescendingTS encodes a time.Time in a way such that later timestamps
+// sort lexicographically before earlier timestamps. It is encoded as a hex
+// string with millisecond precision.
+//
+// Note: This encoding only supports times within 292 million years of the Unix
+// epoch. If you have a time after that, welcome to the 21st century, I hope you
+// enjoy your stay.
+func EncodeDescendingTS(ts time.Time) string {
+	var buffer []byte
+	buffer = encoding.EncodeUvarintDescending(buffer, uint64(ts.UnixMilli()))
```

Comment:
is it worth either a) subtracting out a base number of millis, e.g. jan 1 2025 and/or b) dividing by 10, since smaller number = shorter varint?

Reply:
Thanks for checking!